pkg/operator/controller/util.go

```diff
@@ -46,3 +50,155 @@ func mergeLabelsAndAnnotations(src, dest metav1.Object) {
 func deployClusterResources() bool {
 	return strings.ToLower(os.Getenv("DEPLOY_CLUSTER_RESOURCES")) != "false"
 }
+
+// MarkCrHealthyMessage marks the passed in CR as healthy. The CR object needs to be updated by the caller afterwards.
+// Healthy means the following status conditions are set:
+// ApplicationAvailable: true
+// Progressing: false
+// Degraded: false
+func MarkCrHealthyMessage(cr *cdiv1alpha1.CDI, reason, message string) {
+	conditions.SetStatusCondition(&cr.Status.Conditions, conditions.Condition{
+		Type:    conditions.ConditionAvailable,
+		Status:  corev1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+	conditions.SetStatusCondition(&cr.Status.Conditions, conditions.Condition{
+		Type:   conditions.ConditionProgressing,
+		Status: corev1.ConditionFalse,
+	})
+	conditions.SetStatusCondition(&cr.Status.Conditions, conditions.Condition{
+		Type:   conditions.ConditionDegraded,
+		Status: corev1.ConditionFalse,
+	})
+}
+
+// MarkCrHealthy marks the passed in CR as healthy. The CR object needs to be updated by the caller afterwards.
+// Healthy means the following status conditions are set:
+// ApplicationAvailable: true
+// Progressing: false
+// Degraded: false
+func MarkCrHealthy(cr *cdiv1alpha1.CDI) {
```

Comment:
why not have this call `MarkCrHealthyMessage`?

Reply:
I couldn't figure out how to make it update the status but not overwrite the message any other way.

Reply:
The way I read the code, the reason/message will be set to `""` wheen `MarkCrHealthy` is called.  But maybe I'm wrong.